clang/test/Analysis/Checkers/WebKit/uncounted-local-vars.cpp

```diff
@@ -289,3 +289,32 @@ void foo() {
 }
 
 } // namespace local_assignment_to_global
+
+namespace local_var_in_recursive_function {
+
+struct TreeNode {
+  Ref<TreeNode> create() { return Ref(*new TreeNode); }
+
+  void ref() const { ++refCount; }
+  void deref() const {
+    if (!--refCount)
+      delete this;
+  }
+
+  int recursiveCost();
+
+  int cost { 0 };
+  mutable unsigned refCount { 0 };
+  TreeNode* nextSibling { nullptr };
+  TreeNode* firstChild { nullptr };
+};
+
+int TreeNode::recursiveCost() {
+  // no warnings
+  unsigned totalCost = cost;
+  for (TreeNode* node = firstChild; node; node = node->nextSibling)
+    totalCost += recursiveCost();
+  return totalCost;
+}
+
+} // namespace local_var_in_recursive_function
```

Comment:
Can we also have a test for a recursive function that isn't trivial? say, a recursive function calling a non-trivial method?

Reply:
https://github.com/llvm/llvm-project/blob/main/clang/test/Analysis/Checkers/WebKit/uncounted-obj-arg.cpp has a whole bunch of non-trivial recursive functions but we can add one here for alpha.webkit.UncountedLocalVarsChecker as well.

Reply:
Added a test case.